File: 2024/day01_2.py
# pylint: disable=missing-module-docstring, missing-function-docstring


from typing import Any


def solution(content: list[str]) -> Any:
    split = [line.split() for line in content]
    right = [a for *_, a in split]
    return sum(int(a) * len([r for r in right if r == a]) for a, *_ in split)


def main() -> None:
    with open("day01.dat", "r", encoding="utf8") as file:
        print(solution(file.readlines()))


if __name__ == "__main__":
    main()

File: 2024/day01-2.ts
function solution(content: string) {
  const split = content
    .split("\n")
    .map((line) => line.split("   ").map((item) => parseInt(item)));
  const left = split.map((item) => item.at(0)!);
  const right = split.map((item) => item.at(1)!);
  return left
    .map((l) => l * right.filter((r) => r === l).length)
    .reduce((acc, cur) => acc + cur, 0);
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
